metacat-common-server/src/main/java/com/netflix/metacat/common/server/connectors/AuthEnabledConnectorTableService.java

```diff
@@ -0,0 +1,125 @@
+package com.netflix.metacat.common.server.connectors;
+
+import com.netflix.metacat.common.MetacatRequestContext;
+import com.netflix.metacat.common.QualifiedName;
+import com.netflix.metacat.common.dto.Pageable;
+import com.netflix.metacat.common.dto.Sort;
+import com.netflix.metacat.common.exception.MetacatUnAuthorizedException;
+import com.netflix.metacat.common.server.api.authorization.Authorization;
+import com.netflix.metacat.common.server.api.authorization.AuthorizationStatus;
+import com.netflix.metacat.common.server.connectors.model.TableInfo;
+import com.netflix.metacat.common.server.util.MetacatContextManager;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Connector that authorizes requests based on the request context.
+ */
+@Slf4j
+@RequiredArgsConstructor
+public class AuthEnabledConnectorTableService implements ConnectorTableService {
+    @Getter
+    @NonNull
+    private final ConnectorTableService delegate;
+    @NonNull
+    private final Authorization authorization;
+
+    @Override
+    public TableInfo get(final ConnectorRequestContext context,
+                         final QualifiedName name) {
+        authorize(MetacatContextManager.getContext(), name);
+        return delegate.get(context, name);
+    }
+
+    @Override
+    public Map<String, List<QualifiedName>> getTableNames(final ConnectorRequestContext context,
+                                                          final List<String> uris,
+                                                          final boolean prefixSearch) {
+        return delegate.getTableNames(context, uris, prefixSearch);
+    }
+
+    @Override
+    public List<QualifiedName> getTableNames(final ConnectorRequestContext context,
+                                             final QualifiedName name,
+                                             final String filter,
+                                             final Integer limit) {
+        authorize(MetacatContextManager.getContext(), name);
```

Comment:
Now that you're adding a new `AuthEnabledConnector` services in addition to existing `ThrottlingConnector` ones there is an opportunity for some refactor and improvement. Instead of overriding each method similar to what the throttling connector already does, there could be a parent class that does these overrides and calls an abstract `check` and both auth and throttling provide their own implementation of this.

Reply:
That being the case, this update could also be done in a later change.

Reply:
addressed all comments in a refactor to combine auth and throttling into one validating class 😄